### DISCUSSION TITLE
Confused by inconsistent behaviour with AutoIncrementing Primary Key

### DISCUSSION THREAD

**Discussion by CharlieArmstrongDev** [ServiceStack/Discuss#27] [Q&A] [answered]
Hi, I am very early on with my attempt to actually link a database into my ServiceStack project and create the DTOs and write services etc. I used AutoQuery with the GenerateCrudServices AutoRegister feature at first to make every endpoint straight away, but then copied the ones I need at first into their own files in ServiceModel and turned it off.

In one case, this was the original table:
![image](https://user-images.githubusercontent.com/97681489/186659420-97aef940-f90d-4510-a68b-acad779fd29f.png)
and to begin with the POST DTO looked like this but without the Id field which I had to add as sending a request on the API explorer without it would just send an error that Primary Key was required:
![image](https://user-images.githubusercontent.com/97681489/186662281-aee71689-cfc8-496d-babf-6c056760720a.png)

So technically you can choose the Id which shouldn't be allowed, but I noticed if you input the Id as 0 then it acts as it should and actually sets the Id to be the next highest value (although still returning Id:0 in the response) which seemed like a good enough workaround for now.

However, I have tried to do the same thing on another table:
![image](https://user-images.githubusercontent.com/97681489/186664897-3e336d78-0008-40c4-bbbc-4842618d47ff.png)
![image](https://user-images.githubusercontent.com/97681489/186665273-1107c596-0b7a-44a5-b3b0-5a8e0045e025.png)
Again I added the AutoIncrement Id field to what was generated in the API explorer to match the DB table, but in this case it doesn't seem to act the same?

I can specify any (non-duplicate) Id, but if I try to enter Id: 0, it simply gives me an argument exception.
![image](https://user-images.githubusercontent.com/97681489/186665700-35cf2c22-76b1-4bf3-8a9a-2f26fada65ad.png)

If anyone could help me on why they would respond differently to an entry of 0? Or rather what I should do to avoid the workaround entirely?

**Comment by CharlieArmstrongDev** [comment]
I realise the reason why it had been possible was because I had set Id as a long rather than an int as it is in the table, which would allow me to put in 0. Is it intentional that 0 triggers the AutoIncrement while any other number positive or negative will override it?

Any explanation on what is happening at any stage here would be welcome given I am quite lost about what could be happening behind the scenes here and if I feel like what I am doing is a workaround then maybe AutoQuery might not be the way to go? I am not overly familiar with MySQL either.

**Comment by mythz** [comment, marked as answer]
The OrmLite attributes, e.g. `[AutoIncrement]` needs to be on the Data Model (e.g. `Address`) not on the Request DTO.

and if you have an `[AutoIncrement]` PK like `Id` it shouldn't be on the `ICreateDb<T>` Request DTO as the RDBMS should be generating it.
